src/views/List.jsx

```diff
@@ -31,59 +33,89 @@ export function List({ data, listToken }) {
 			.catch(() => setCopied('Not Copied.'));
 	};
 
+	const deleteList = () => {
+		if (
+			window.confirm(
+				'Are you sure you want to delete your shopping list? This cannot be undone.',
+			)
+		) {
+			const itemsToBeDeleted = [];
+			data.forEach((item) => {
+				itemsToBeDeleted.push(deleteItem(listToken, item.id));
+			});
+			Promise.all(itemsToBeDeleted)
```

Comment:
Great use of Promise.all to make sure everything is deleted before calling setListToken again without needing a timeout. And `deleteList` is a very satisfying UX action that definitely has a use case (+1 on including in a "no item" list scenario).